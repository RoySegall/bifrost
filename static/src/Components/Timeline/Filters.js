import React from 'react';
import {Calendar, Car, Plane, Hotel, Friends} from '../../Fonts'


export default class Filters extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            filters: [
                {'key': 'all', title: 'All events', icon: Calendar(), first: true},
                {'key': 'pickingcarSet', title: 'Picking car', icon: Car()},
                {'key': 'flightSet', title: 'Flight', icon: Plane()},
                {'key': 'accommodationSet', title: 'Accommodation', icon: Hotel()},
                {'key': 'meetingconjunctionSet', title: 'Meet employees', icon: Friends()},
            ],
            activeFilter: 'all',
        };
    }

    switchFilter(filter) {
        this.setState({activeFilter: filter});
    }

    List() {
        return this.state.filters.map((item, key) => {

            const active = this.state.activeFilter === item['key'] ? 'selected' : '';
            const first = item['first'] ? 'first' : '';

            return <li
                key={key}
                className={`list-inline-item filter ${item['key']} ${active} ${first}`}
                onClick={() => this.switchFilter(item['key'])}>
                <span>{item['icon']} {item['title']}</span>
            </li>
        })
    }

    render() {
        return <div className="row">

            <div className="col-12 filters">
                <ul className="filters float-left">{this.List()}</ul>
            </div>

        </div>
    }
}
