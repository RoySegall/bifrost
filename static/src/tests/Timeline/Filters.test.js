import React from 'react';
import renderer from 'react-test-renderer';
import Filters from '../../Components/Timeline/Filters';
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

const filterCallbackMock = jest.fn();

test("Testing the snapshot of the component", () => {

    const component = renderer.create(<Filters/>);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("Testing the filter changing", () => {

    const filters = mount(<Filters changeEvnetCallbak={filterCallbackMock}/>);
    const event = {
        preventDefault: () => {
        }
    };

    // Mock the event for the the onclick button.
    jest.spyOn(event, 'preventDefault');

    // Verify we got the filters.
    expect(filters.find('.fal')).toHaveLength(5);

    // Click on the button and verify the event was triggered.
    filters.find('button.pickingCars').simulate('click');
    expect(filters.instance().state.activeFilter).toBe('pickingCars');
    expect(filterCallbackMock).toHaveBeenCalledTimes(1);

    filters.find('button.flights').simulate('click');
    expect(filters.instance().state.activeFilter).toBe('flights');
    expect(filterCallbackMock).toHaveBeenCalledTimes(2);

    filters.find('button.accommodations').simulate('click');
    expect(filters.instance().state.activeFilter).toBe('accommodations');
    expect(filterCallbackMock).toHaveBeenCalledTimes(3);

    filters.find('button.meetingConjunctions').simulate('click');
    expect(filters.instance().state.activeFilter).toBe('meetingConjunctions');
    expect(filterCallbackMock).toHaveBeenCalledTimes(4);

    filters.find('button.all').simulate('click');
    expect(filters.instance().state.activeFilter).toBe('all');
    expect(filterCallbackMock).toHaveBeenCalledTimes(5);
});
