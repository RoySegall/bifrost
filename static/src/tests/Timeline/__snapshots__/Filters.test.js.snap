// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the snapshot of the component 1`] = `
<div>
  <ul
    className="flex flex-row justify-start pt-4"
  >
    <li
      onClick={[Function]}
    >
      <button
        className="
            btn-filter-selected font-bold pt-2 pb-2 pl-4 pr-4 ml-2 rounded
          "
      >
        <span>
          <i
            className="fal fa-calendar-alt"
          />
        </span>
        <span
          className="pl-2"
        >
          All events
        </span>
      </button>
    </li>
    <li
      onClick={[Function]}
    >
      <button
        className="
            btn-filter font-bold pt-2 pb-2 pl-4 pr-4 ml-2 rounded
          "
      >
        <span>
          <i
            className="fal fa-car"
          />
        </span>
        <span
          className="pl-2"
        >
          Picking car
        </span>
      </button>
    </li>
    <li
      onClick={[Function]}
    >
      <button
        className="
            btn-filter font-bold pt-2 pb-2 pl-4 pr-4 ml-2 rounded
          "
      >
        <span>
          <i
            className="fal fa-plane"
          />
        </span>
        <span
          className="pl-2"
        >
          Flight
        </span>
      </button>
    </li>
    <li
      onClick={[Function]}
    >
      <button
        className="
            btn-filter font-bold pt-2 pb-2 pl-4 pr-4 ml-2 rounded
          "
      >
        <span>
          <i
            className="fal fa-hotel"
          />
        </span>
        <span
          className="pl-2"
        >
          Accommodation
        </span>
      </button>
    </li>
    <li
      onClick={[Function]}
    >
      <button
        className="
            btn-filter font-bold pt-2 pb-2 pl-4 pr-4 ml-2 rounded
          "
      >
        <span>
          <i
            className="fal fa-user-friends"
          />
        </span>
        <span
          className="pl-2"
        >
          Meet employees
        </span>
      </button>
    </li>
  </ul>
</div>
`;
